Explorer/Assets/DCL/Chat/ChatInputBoxElement.cs

```diff
@@ -0,0 +1,447 @@
+        {
+            if (suggestionPanel.IsActive)
+            {
+                suggestionPanelController!.SetPanelVisibility(false);
+                lastMatch = Match.Empty;
+                return;
+            }
+
+            if (emojiPanel.gameObject.activeInHierarchy)
+            {
+                emojiPanelButton.SetState(false);
+                emojiPanelController!.SetPanelVisibility(false);
+                EmojiSelectionVisibilityChanged?.Invoke(false);
+            }
+
+            if (string.IsNullOrWhiteSpace(submittedText))
+            {
+                inputField.DeactivateInputField();
+                inputField.OnDeselect(null);
+                return;
+            }
+
+            if (chatAudioSettings.chatAudioSettings == ChatAudioSettings.ALL)
+                UIAudioEventsBus.Instance.SendPlayAudioEvent(chatSendMessageAudio);
+
+            inputField.ResetInputField();
+
+            InputSubmitted?.Invoke(submittedText, ORIGIN);
+        }
+
+        public void Dispose()
+        {
+            if (emojiPanelController != null)
+            {
+                emojiPanelController.EmojiSelected -= OnEmojiSelected;
+                emojiPanelController.Dispose();
+            }
+
+            if (suggestionPanelController != null)
+            {
+                suggestionPanelController.Dispose();
+                suggestionPanelController.SuggestionSelected -= OnSuggestionSelected;
+            }
+
+            emojiPanelCts.SafeCancelAndDispose();
+
+            inputField.OnRightClickEvent -= OnRightClickRegistered;
+            inputField.OnPasteShortcutPerformedEvent -= OnPasteShortcutPerformed;
+        }
+
+        private void OnSuggestionSelected(string suggestionId)
+        {
+            ReplaceSuggestionInText(suggestionId);
+        }
+
+        private void ReplaceSuggestionInText(string suggestion)
+        {
+            if (!lastMatch.Success || !inputField.IsWithinCharacterLimit(suggestion.Length - lastMatch.Groups[1].Length)) return;
+
+            UIAudioEventsBus.Instance.SendPlayAudioEvent(addEmojiAudio);
+            int replaceAmount = lastMatch.Groups[1].Length;
+            int replaceAt = wordMatchIndex + lastMatch.Groups[1].Index;
+
+            inputField.ReplaceTextAtPosition(replaceAt, replaceAmount, suggestion);
+
+            inputField.ActivateInputField();
+
+            lastMatch = Match.Empty;
+        }
+
+        private void UpdateProfileNameMap()
+        {
+            //NOTE: This information should come from the channel where this chat is taking place and that channel should make sure this list is updated.
+            //For now this will work, but is not the final implementation, this will be changed in the next shape.
+            IReadOnlyCollection<string> remoteParticipantIdentities = viewDependencies.RoomHub.IslandRoom().Participants.RemoteParticipantIdentities();
+
+            var profileSuggestions = profileSuggestionsDictionary.ToList();
```

Comment:
Use a pooled list instead as you invoke this function on input change (frequently)